### PR TITLE
Improve filterFeatures() solve issue #150

### PR DESCRIPTION
**Pull request by cvanderaa**
(please first merge #165)

This PR:

- Improves docs for filterFeatures()
- Improves unit tests for filterFeatures()
- Adds a `keep` argument in filterFeatures() to control whether to keep or remove features for assays that do not contain the filter variable. Also added message printing for a better overview of which variable were found.

### CODE REVIEW

**Comment by cvanderaa** [comment]
This PR is ready for review.

File: R/QFeatures-filter.R [hunk 1]
```diff
@@ -7,7 +7,29 @@
...
-##'
+##' 
+##' @section The filtering procedure:
+##' 
+##' `filterFeatures()` will go through each assay of the `QFeatures` 
+##' object and apply the filtering on the corresponding `rowData`. 
+##' Features that do not pass the filter condition are removed from 
+##' the assay. In some cases, one may want to filter for a variable 
+##' present in some assay, but not in other. There are two options: 
+##' either provide `"keep = FALSE"` to remove all features for those
```
Comment (lgatto) [inline R/QFeatures-filter.R]: Please drop the quote (it isn't a character) and simply use `keep = FALSE`.

File: R/QFeatures-filter.R [hunk 2]
```diff
@@ -7,7 +7,29 @@
...
+##' 
+##' @section The filtering procedure:
+##' 
+##' `filterFeatures()` will go through each assay of the `QFeatures` 
+##' object and apply the filtering on the corresponding `rowData`. 
+##' Features that do not pass the filter condition are removed from 
+##' the assay. In some cases, one may want to filter for a variable 
+##' present in some assay, but not in other. There are two options: 
+##' either provide `"keep = FALSE"` to remove all features for those 
+##' assays, or provide `"keep = TRUE"` to ignore filtering for those
```
Comment (lgatto) [inline R/QFeatures-filter.R]: as above

File: R/QFeatures-filter.R (hunk 1 above)
Comment (lgatto) [inline R/QFeatures-filter.R]: May be, to be extra clear:

> remove all features for those assays (and thus leaving and empty assay)

File: R/QFeatures-filter.R [hunk 3]
```diff
@@ -197,6 +224,11 @@ VariableFilter <- function(field,
 ##'
 ##' @param na.rm `logical(1)` indicating whether missing values should
 ##'     be removed. Default is `FALSE`.
+##' 
+##' @param keep `logical(1)` indicating whether to keep the features 
+##'     of assays for which at least one of the filtering variables are
+##'     missing in the rowData. When FALSE (default), all such assay 
+##'     will contain 0 features; when TRUE, the assays are untouched.
```
Comment (lgatto) [inline R/QFeatures-filter.R]: Add backticks around TRUE and FALSE

**Review by lgatto — CHANGES_REQUESTED** [review]

**Comment by cvanderaa** [comment]
Sorry, I didn't mean to request a new review. The changes are commited!

**Pull request merged**